### PR TITLE
[refactor] dmd.root.array: Call memset directly from pop() when stomping enabled

### PR DESCRIPTION
**Pull request by ibuclaw**
`Array.remove` calls `memmove`+`memset`, as this is the last element of the array, the `memmove` is useless.

Ran unittests with `ENABLE_DEBUG=1`.  Though for some reason can't run the testsuite with that flag.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/root/array.d [hunk 1]
```diff
@@ -292,8 +292,8 @@ public:
         debug (stomp)
         {
             assert(length);
-            auto result = data[length - 1];
-            remove(length - 1);
+            auto result = data[--length];
+            memset(data.ptr + length, 0xFF, T.sizeof);
             return result;
         }
         else
```
Comment (PetarKirov) [inline src/dmd/root/array.d:299]:
```suggestion
assert(length);
auto result = data[--length];
debug (stomp) memset(data.ptr + length, 0xFF, T.sizeof);
return result;
```

**Comment by MoonlightSentinel** [comment]
> Ran unittests with `ENABLE_DEBUG=1`. Though for some reason can't run the testsuite with that flag.

You don't need to pass that flag to `run.d` as long as the `BUILD` variable points to the right directory (`run.d` and `build.d`  use `release` by default). Or did you encounter another error?

**Comment by WalterBright** [comment]
> is useless

Not really, it tests the edge case in `remove()`. Though that would probably be better done by a better unittest.

`stomp` isn't meant to be efficient.

I'll approve and leave it up to you guys.

**Review by WalterBright — APPROVED** [review]

**Comment by ibuclaw** [comment]
> 
> You don't need to pass that flag to `run.d` as long as the `BUILD` variable points to the right directory (`run.d` and `build.d`  use `release` by default). Or did you encounter another error?
> 
> EDIT: Note that `ENABLE_DEBUG` does not affect `BUILD` (which is a very surprising behaviour inherited from the makefiles...)

If I build with `ENABLE_DEBUG=1`, it builds into generated/debug. From the test directory, it always tries to run the generated/release compiler.

**Comment by ibuclaw** [comment]
> > is useless
> 
> Not really, it tests the edge case in `remove()`. Though that would probably be better done by a better unittest.
> 
> `stomp` isn't meant to be efficient.
> 
> I'll approve and leave it up to you guys.

Just got tripped up momentarily when quickly reviewing the implementation (I ended up not using pop/push to do the thing).

Not bothered either way on my end.

**Comment by MoonlightSentinel** [comment]
> If I build with `ENABLE_DEBUG=1`, it builds into generated/debug.

Then you must have set `BUILD` in your environment because `ENABLE_DEBUG` only affects the `DFLAGS` (search for `ENABLE_DEBUG` in `build.d`)

> From the test directory, it always tries to run the generated/release compiler.

Did you specify `BUILD=debug`?
`src/build.d BUILD=debug && test/run.d BUILD=debug` builds and tests `generated/linux/debug/64/dmd` on my end.
